src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexThresholds.cs

```diff
@@ -22,12 +22,18 @@ internal static class SymbolicRegexThresholds
         /// processing starts out in DFA mode, even if we've previously triggered NFA mode for the same regex.
         /// We switch over into NFA mode the first time a given traversal (match operation) results in us needing
         /// to create a new node and the graph is already or newly beyond this threshold.
+        /// TODO: summarize this
+        /// this should be a very last resort action, going from DFA mode to NFA mode turns 500MB/s to 5MB/s
+        /// with an entirely different search-time algorithmic complexity
+        /// 100_000 isn't a really a high memory cost either,
+        /// ideally NFA mode should never be used, 1_000_000 is ok as well but it depends how much memory the user has
         /// </remarks>
-        internal const int NfaThreshold = 10_000;
+        internal const int NfaThreshold = 100_000;
 
         /// <summary>
         /// Default maximum estimated safe expansion size of a <see cref="SymbolicRegexNode{TSet}"/> AST
         /// after the AST has been anlayzed for safe handling.
+        /// TODO: this is perhaps too conservative, consider raising this, 5000 is ok even in safety critical scenarios, ~50 000 for general purpose is ok too
```

Comment:
What limit would you recommend? @veanes?

Reply:
But I also just remembered that regex size limit is also configurable by using an environment variable. 
The NfaThreshold limit is not configurable. Well, it is an internal algorithm parameter, so as such, not sure if it should be exposed through an environment variable, although ideally this could be useful in cases where it is known that the DFA can be large but is also simple/benign.